Make block device allocation settings configurable

nkrinner opened the pull request:
Make block_device_allocate_retries and block_device_allocate_retries_interval
from the crowbar web ui.
See also https://bugzilla.suse.com/show_bug.cgi?id=937800

mjura committed:
Multi Ceph backend integration improvements

Conflicts:
	chef/cookbooks/nova/recipes/ceph.rb
	chef/cookbooks/nova/recipes/config.rb

mjura committed:
Change system calls on Chef style

mjura committed:
After secret definition delete secret file

mjura committed:
Import code from ceph barclamp

mjura committed:
Fix export of the nova admin key to libvirt

This patch was backported from
https://github.com/crowbar/barclamp-ceph/pull/84/

nkrinner committed:
Split recipes in nova-compute-hyperv role

We split the the setup recipe in the hyperv cookbook in two seperate
recipes do_setup and do_nova to make the basic setup recipe usable for
other cookbooks too.

mjura committed:
Import client_key from rbd user keyring file

vuntz committed:
Drop PFS support

Pull from source hasn't been maintained for quite some time and it's
very likely broken. So drop support for it.

vuntz committed:
Do not forcefully install packages on Debian/Ubuntu

This was probably for some very old packaging bug

vuntz committed:
Minor cleanup

No need to specify :install for package resources as that is the
default.

vuntz committed:
Merge pull request #450 from vuntz/kill-pfs

[post-tex] Drop PFS support (and some cleanup)

Commit:
Drop ProposalObject and use Proposal model instead

mjura committed:
Avoid creating secret xml file during each chef run

mjura committed:
Merge pull request #452 from bkutil/drop_proposal

Drop ProposalObject and use Proposal model instead

mjura committed:
Merge pull request #451 from nkrinner/split-hyperv-recipes

Do not merge yet - Split recipes in nova-compute-hyperv role

mjura committed:
Disable default[:nova][:rbd] attributes in nova.conf

mjura committed:
Merge pull request #449 from mjura/multi_ceph_backends

Multi Ceph backend integration improvements

vuntz committed:
Fix broken detection of platform_family

node.platform_family? just doesn't work. The test should either just be
platform_family? or on node[:platform_family].

vuntz committed:
Allow deploying keystone on SLE12 nodes

vuntz committed:
Merge pull request #454 from vuntz/sle12

Allow deploying nova-controller on SLE12 nodes

vuntz committed:
ceph: Fix parsing of "virsh secret-list" header (bsc#931284)

The first line may start with "UUID" or " UUID". We were not supporting
the latter...

https://bugzilla.suse.com/show_bug.cgi?id=931284

vuntz committed:
Merge pull request #458 from vuntz/secret-list-parsing

ceph: Fix parsing of "virsh secret-list" header

MaximilianMeister committed:
Add support for nova.virt.zvm.ZVMDriver

rhafer committed:
Allow using the Keystone v3 API

The crowbar-nova-set-availability-zone tool is still lacking support for
the v3 API. The same is true for the way how nova uses the neutronclient. We
hardcode the use of the v2 in both cases.

vuntz committed:
Enable the 2.1 API (bsc#934225)

Quite confusingly, the 2.1 API is disabled by default and only enabled
when this is in nova.conf:

[osapi_v3]
enabled=true

Since we configure the computev21 endpoint unconditionally, also enable
the API.

Fix found by Roberto Sassu <rsassu@suse.com>

https://bugzilla.suse.com/show_bug.cgi?id=934225

dirkmueller committed:
Merge pull request #462 from vuntz/v21-enabled

Enable the 2.1 API (bsc#934225)

Commit:
Added badge for gitter chat to README

Commit:
Added more contact informations to README

Commit:
Mention SUSE as the maintainer within README

Commit:
Merge pull request #464 from tboerger/feature/gitter-badge

Improve README, add badge, contact and SUSE maintainer

Commit:
Fixed gitter link to a public available channel

rhafer committed:
Make max_header_line configurable

When using the v3 API auth tokes can grow larger than with the v2 API. The
default value should normally work but making this adjustable should help in
situation where a lot of endpoints are defined.

rhafer committed:
Make max_header_line configurable

When using the v3 API auth tokes can grow larger than with the v2 API. The
default value should normally work but making this adjustable should help in
situation where a lot of endpoints are defined.

(cherry picked from commit 94b16d372dfcc77c1479d67c59d183a82e324fe7)
Signed-off-by: Ralf Haferkamp <rhafer@suse.de>

dirkmueller committed:
Merge pull request #461 from rhafer/v3

Allow using the Keystone v3 API

Commit:
Introduce HoundCI

We want to test HoundCI, this configs are needed to run Hound with
our preferred styles.

Commit:
Merge pull request #465 from rsalevsky/hound

Introduce HoundCI

Commit:
Fix typo in rubocop config

Commit:
Merge pull request #466 from rsalevsky/fix_hound

Fix typo in rubocop config

mjura committed:
Make nova_floating independent of public network

Network notes don't need a public IP allocated,
but they need an interface in the VLAN for nova_floating enabled.

MaximilianMeister committed:
Use 2 (not 4) spaces for indentation

dirkmueller committed:
Merge pull request #460 from MaximilianMeister/zvm-support

adding zvm nova compute roles

bkutil committed:
Merge pull request #470 from crowbar/feature/chef_db_drop

Feature/chef db drop

toabctl committed:
Set cookbook name in metadata

Needed if the cookbook is used without the crowbar
framework. I.e. usingthe cookbook as berkshelf dependency.
Also update maintainer email address.

toabctl committed:
Merge pull request #471 from toabctl/cookbook-name

Set cookbook name in metadata

dirkmueller committed:
Mark xCat password field as being a password field

otherwise the password is not hidden during typing.

dirkmueller committed:
Merge pull request #472 from dirkmueller/master

Mark xCat password field as being a password field

Commit:
Removed smoketest folder, not used anymore

MaximilianMeister committed:
Merge pull request #473 from tboerger/feature/smoketest-remove

Removed smoketest folder, not used anymore

dirkmueller committed:
Merge pull request #469 from mjura/floating-public-split

Make nova_floating independent of public network

vuntz committed:
Fix catalog_info setting for cinder in nova.conf

It was referring to volumev2:cinder:internalURL, but should be
volumev2:cinderv2:internalURL (as that's what is registered in the
cinder cookbook).

vuntz committed:
Merge pull request #475 from vuntz/cinder-catalog

Fix catalog_info setting for cinder in nova.conf

nkrinner committed:
Make block device allocation settings configurable

Make block_device_allocate_retries and block_device_allocate_retries_interval
configurable from the crowbar web ui in the raw view.
See also https://bugzilla.suse.com/show_bug.cgi?id=937800

jsuchome commented:
Looks good so far. I think if it works it's a way to go.

tboerger commented on chef/data_bags/crowbar/migrate/nova/011_configure_block_device_allocation.rb:
```diff
@@ -0,0 +1,11 @@
+def upgrade ta, td, a, d
+  a['block_device_allocation_retries'] = 60
```
Normally we use `ta["block_device_allocation_retries"]` for assignment.

tboerger commented on chef/data_bags/crowbar/migrate/nova/011_configure_block_device_allocation.rb:
```diff
@@ -0,0 +1,11 @@
+def upgrade ta, td, a, d
+  a['block_device_allocation_retries'] = 60
+  a['block_device_allocation_retries_interval'] = 3
```
Normally we use `ta["block_device_allocation_retries_interval"]` for assignment.

Commit:
increase timeout for synchronization of HA resources (bsc#935462)

Nova HA resources can take over 1 minute to be set up, as seen by
mkcloud failures in ci.suse.de Jenkins.

https://bugzilla.suse.com/show_bug.cgi?id=935462

dirkmueller committed:
Merge pull request #478 from aspiers/ha-sync-timeout

increase timeout for synchronization of HA resources (bsc#935462)

Commit:
If tempest is deployed, load docker image at docker compute node
so that tempest could run tests with docker based image.

The image is packaged in tempest barclamp. We can't do this from
tempest recipe because tempest might be deployed on another node
which does not have access to the docker comput node.

download and load docker image only once

MaximilianMeister committed:
Updated README from master and direct to the new repository

nkrinner commented:
Removed the [WIP] tag from the pull request headline. Unrelated if this fixes the bug, this additional configuration option can get merged to the code. Will also submit against crowbar-openstack when this gets merged.

rsalevsky commented:
+1

MaximilianMeister commented:
+1 codewise

rossella committed:
nova: Correctly configure haproxy for metadata when ssl is on

Backport from https://github.com/crowbar/crowbar-openstack/pull/8

dirkmueller committed:
Add option to enforce use of config drive (bsc#943166, bsc#941531)

This is useful in environments where the dhcp-agent is
disabled, which means there is no neutron/nova metadata proxy
available. Also inject the flat network in that case, so
that cloud-init can pick up network configuration in case
DHCP is turned off.

Backport of https://github.com/crowbar/crowbar-openstack/pull/17

Co-Authored-By: Thomas Bechtold <tbechtold@suse.com>

vuntz committed:
Add note that this got moved to merged repo

vuntz commented on chef/data_bags/crowbar/bc-template-nova.json:
```diff
@@ -98,14 +98,16 @@
-      }
+      },
+      "block_device_allocate_retries": 60,
+      "block_device_allocate_retries_interval": 3
```
Would it make sense to create a new group for the options?

tboerger replied:
Sounds good to me

vuntz commented:
@nkrinner in general, this must go in crowbar-openstack first, then we can merge the backport.

nkrinner commented:
Closing this pr for now, as the discussion is happening in https://github.com/crowbar/crowbar-openstack/pull/39 and this one got messed up during rebasing.
